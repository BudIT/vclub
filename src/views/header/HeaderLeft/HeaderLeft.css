.ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: flex-start;
  background-color: #FFD00B;
  flex: 1;
}

.a {
  color: #33363A;
  text-decoration: none;
  text-align: center;
  padding: 14px 16px;
  font-size: 22px;
  transition: 0.2s;
  display: inline-block;
  font-family: 'Rubik', helvetica, sans-serif;
}

.aCurrent {
  composes: a;
  text-decoration: underline;
}

.a:hover {
  background-color: #FFAA00;
}
