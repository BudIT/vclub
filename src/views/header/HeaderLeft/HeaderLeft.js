import React, { PropTypes } from 'react';

import {
  ChatRoomType, WhiteboardRoomType, SharingRoomType, MediaRoomType,
} from 'vclub/constants/roomTypes';

import HeaderTab from './HeaderTab/HeaderTab';

import style from './HeaderLeft.css';


const roomTitles = {
  [ChatRoomType]: 'Беседка',
  [SharingRoomType]: 'Шеринг',
  [WhiteboardRoomType]: 'Доска',
  [MediaRoomType]: 'Медиа',
};

// changeRoom
function HeaderLeft(props) {
  const {
    roomsNames,
    currentRoomName,
    dispatch,
    user,
  } = props;

  return (
    <ul className={style.ul}>
      {roomsNames.map((roomName, index) => (
        <HeaderTab
          isCurrentTab={roomName === currentRoomName}
          key={index.toString()}
          user={user}
          roomName={roomName}
          dispatch={dispatch}
        >
          {roomTitles[roomName] || roomName}
        </HeaderTab>
      ))}
    </ul>
  );
}

HeaderLeft.propTypes = {
  currentRoomName: PropTypes.string.isRequired,
  roomsNames: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  dispatch: PropTypes.func.isRequired,
  user: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    master: PropTypes.bool.isRequired,
  }).isRequired,
};

export default HeaderLeft;
