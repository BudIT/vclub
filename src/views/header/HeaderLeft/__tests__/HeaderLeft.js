/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import HeaderLeft from '../HeaderLeft';


const roomsNames = ['SHARING', 'MEDIA', 'CHAT', 'WHITEBOARD'];
const currentRoomName = 'MEDIA';

const user = {
  id: '1',
  name: 'User',
  master: true,
};

test('<HeaderLeft /> renders correctly', () => {
  const dispatchSpy = jest.fn();
  const wrapper = shallow(
    <HeaderLeft
      currentRoomName={currentRoomName}
      roomsNames={roomsNames}
      user={user}
      dispatch={dispatchSpy}
    />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
