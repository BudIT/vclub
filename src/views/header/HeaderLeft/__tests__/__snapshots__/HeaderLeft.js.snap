exports[`test <HeaderLeft /> renders correctly 1`] = `
<ul
  className="ul">
  <HeaderTab
    dispatch={[Function]}
    isCurrentTab={false}
    roomName="SHARING"
    user={
      Object {
        "id": "1",
        "master": true,
        "name": "User",
      }
    }>
    Шеринг
  </HeaderTab>
  <HeaderTab
    dispatch={[Function]}
    isCurrentTab={true}
    roomName="MEDIA"
    user={
      Object {
        "id": "1",
        "master": true,
        "name": "User",
      }
    }>
    Медиа
  </HeaderTab>
  <HeaderTab
    dispatch={[Function]}
    isCurrentTab={false}
    roomName="CHAT"
    user={
      Object {
        "id": "1",
        "master": true,
        "name": "User",
      }
    }>
    Беседка
  </HeaderTab>
  <HeaderTab
    dispatch={[Function]}
    isCurrentTab={false}
    roomName="WHITEBOARD"
    user={
      Object {
        "id": "1",
        "master": true,
        "name": "User",
      }
    }>
    Доска
  </HeaderTab>
</ul>
`;
